src/main/proto/game_data.proto

```diff
@@ -0,0 +1,158 @@
+syntax = "proto2";
+ 
+package rlbot.api;
+
+message ControllerState {
+  optional float throttle = 1; // -1 for full reverse, 1 for full forward
+  optional float steer = 2; // -1 for full left, 1 for full right
+  optional float pitch = 3; // -1 for nose down, 1 for nose up
+  optional float yaw = 4; // -1 for full left, 1 for full right
+  optional float roll = 5; // -1 for roll left, 1 for roll right
+  optional bool jump = 6;
+  optional bool boost = 7;
+  optional bool handbrake = 8;
+}
+
+message PlayerInput {
+  required int32 player_index = 1;
```

Comment:
proto fields should not be required.
Especially if we ever plan on writing them to a file or making backwards compatible changes in the future

Reply:
I'm reluctant to make everything optional because I feel like we'd need to do an unreasonable number of checks in c++ to avoid crashes.

Reply:
we can perform those checks client side.  but making it optional is safer on the wire.  if we remove a field at a later point then everything written to disk with that message (or between dll versions) become incompatible

Reply:
OK, fair enough. Does that advice apply to very primitive things like Vector3 and Rotator, or do you like those the way they are with the *Partial counterparts?

Reply:
generally it is supposed to apply to everything.
But I really doubt we would change the makeup of Vector3 or Rotator so I am fine with leaving those as required.